### PR TITLE
feat: Added a donate menu, and optimized email verification and messaging.

### PR DESCRIPTION
**Pull request by frankoslive**

### CODE REVIEW

File: src/commander/common.ts [hunk 1]
```diff
@@ -50,10 +50,10 @@ export async function updateMenu(versions) {
     },
   ];
   console.log(
-    '-----------------------------------------------\n' +
-      `Client Current Version: ${versions.current}\n` +
+    `${Font.fgCyan}${Font.bright}-----------------------------------------------\n` +
+      `Client Current Version:${Font.reset}${Font.bright} ${versions.current}${Font.reset}\n` +
```
Comment (RobinHolmes) [inline src/commander/common.ts:54]: add a space

File: src/commander/synchronizer.ts [hunk 2]
```diff
@@ -104,13 +117,20 @@ export class SynchronizerCommander {
...
+      set_donation_ratio: async () => await this.setDonationRatio(),
       purchase_memory_slot: async () => await this.purchaseSlots(),
       reset_btc_rpc: async () => await this.resetBtcRpcUrl(),
       export_private_key: async () => {
         console.log(`Private Key:${this.exsatAccountInfo.privateKey}`);
         await input({ message: 'Press [enter] to continue' });
       },
       remove_account: async () => await this.removeKeystore(),
+      change_email: async () => {
+        console.log();
```
Comment (RobinHolmes) [inline src/commander/synchronizer.ts:129]: console.log();  ?

File: src/commander/synchronizer.ts [hunk 3]
```diff
@@ -104,13 +117,20 @@ export class SynchronizerCommander {
...
+        await changeEmail(accountName, this.exsatAccountInfo.email);
+        console.log();
```
Comment (RobinHolmes) [inline src/commander/synchronizer.ts:131]: ?

File: src/commander/synchronizer.ts [hunk 4]
```diff
@@ -189,6 +210,29 @@ export class SynchronizerCommander {
     return true;
   }
 
+  async setDonationRatio() {
+    const ratio = await inputWithCancel('Enter Donation Ratio(0-10000,Input "q" to return):', (value) => {
```
Comment (RobinHolmes) [inline src/commander/synchronizer.ts:214]: Add a space before 'Input'

File: src/commander/synchronizer.ts [hunk 5]
```diff
@@ -266,7 +312,7 @@ export class SynchronizerCommander {
     if (rpcUrl) {
       if (
         !(await confirm({
-          message: `Your BTC_RPC_URL:${rpcUrl}\nAre you sure to reset it?`,
+          message: `Your BTC_RPC_URL:${rpcUrl}\nAre you sure to change it?`,
```
Comment (RobinHolmes) [inline src/commander/synchronizer.ts:315]: Add a space after ':'

File: src/commander/common.ts (hunk 1 above)
Comment (frankoslive) [reply src/commander/common.ts]: Done

File: src/commander/synchronizer.ts (hunk 4 above)
Comment (frankoslive) [reply src/commander/synchronizer.ts]: Done

File: src/commander/synchronizer.ts (hunk 5 above)
Comment (frankoslive) [reply src/commander/synchronizer.ts]: Done

File: src/commander/synchronizer.ts (hunk 3 above)
Comment (frankoslive) [reply src/commander/synchronizer.ts:131]: To output a newline

**Review by RobinHolmes — APPROVED** [review]

**Pull request merged**